datafusion/sql/src/expr/function.rs

```diff
@@ -72,6 +73,24 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // next, scalar built-in
         if let Ok(fun) = BuiltinScalarFunction::from_str(&name) {
             let args = self.function_args_to_expr(args, schema, planner_context)?;
+
+            // Translate array_aggregate to aggregate function with array argument.
+            if fun == BuiltinScalarFunction::ArrayAggregate {
+                let fun = match &args[1] {
+                    Expr::Literal(ScalarValue::Utf8(Some(name))) => match name.as_str() {
+                        "sum" => BuiltinScalarFunction::ArraySum,
+                        _ => {
+                            return not_impl_err!(
+                                "Aggregate function {name} is not implemented"
+                            )
+                        }
+                    },
+                    _ => return internal_err!("Aggregate function name is not a string"),
+                };
```

Comment:
It would be better to use `enum` to represent these aggregate functions, like `sum`, and easy to extend later 🤔  But we could do it later pr.